diff --git a/tools/build/src/engine/build.sh b/tools/build/src/engine/build.sh
index 29d08fcbe..48fefb8e6 100755
--- a/tools/build/src/engine/build.sh
+++ b/tools/build/src/engine/build.sh
@@ -180,7 +180,7 @@ check_toolset ()
     fi
     # Intel oneAPI (intel-linux)
     if test_toolset intel-linux && test_path icpx ; then
-        if test_compiler icpx -x c++ -std=c++11 ; then B2_TOOLSET=intel-linux ; return ${TRUE} ; fi
+        if test_compiler icpx -x c++ -std=c++11 -pthread; then B2_TOOLSET=intel-linux ; return ${TRUE} ; fi
     fi
     if test_toolset intel-linux && test_path icc ; then
         if test_compiler icc -x c++ -std=c++11 ; then B2_TOOLSET=intel-linux ; return ${TRUE} ; fi
